tarofreighter/chain_porter.go

```diff
@@ -224,47 +224,66 @@ func (p *ChainPorter) taroPorter() {
 			// reach the state that we broadcast the transaction
 			// that completes the transfer.
 			advancedPkg, err := p.advanceStateUntil(
-				&sendPkg, req.respChan, SendStateBroadcast,
+				&sendPkg, req.respChan, SendStateWaitTxConf,
 			)
 			if err != nil {
 				log.Warnf("unable to advance state machine: %v", err)
 				req.errChan <- err
 				continue
 			}
 
-			// Now that we broadcast the transaction, we'll
-			// create a goroutine that'll wait for the confirmation
-			// then update everything on disk.
+			// The remaining state transitions are evaluated using
+			// a goroutine because tx confirmation and proof
+			// transfer could take a while.
 			p.Wg.Add(1)
-			go p.waitForPkgConfirmation(advancedPkg.OutboundPkg)
```

Comment:
What about just integrating this into the main state machine? So then `advanceStateUntil` wouldn't return until things have proceeded all the way. We'd then adopt an architecture similar the chain planter, where the main goroutine does some prep and book keeping, then we spawn off another struct that finalizes things in a new goroutine.

Reply:
In the latest commits, waiting for tx confirmation has now been moved into the `advanceState` function.